src/main/java/com/google/firebase/auth/internal/FirebaseTokenVerifier.java

```diff
@@ -45,45 +45,61 @@
           + "securetoken@system.gserviceaccount.com";
-  /** The default public keys manager for verifying projects use the correct public key. */
-  public static final GooglePublicKeysManager DEFAULT_KEY_MANAGER =
-      new GooglePublicKeysManager.Builder(new NetHttpTransport.Builder().build(), new GsonFactory())
-          .setClock(Clock.SYSTEM)
-          .setPublicCertsEncodedUrl(CLIENT_CERT_URL)
-          .build();
-
-  private static final String ISSUER_PREFIX = "https://securetoken.google.com/";
+  static final String ID_TOKEN_ISSUER_PREFIX = "https://securetoken.google.com/";
+
+  static final String SESSION_COOKIE_CERT_URL =
+      "https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys";
+  static final String SESSION_COOKIE_ISSUER_PREFIX = "https://session.firebase.google.com/";
+
   private static final String FIREBASE_AUDIENCE =
       "https://identitytoolkit.googleapis.com/google.identity.identitytoolkit.v1.IdentityToolkit";
+
   private static final String ERROR_INVALID_CREDENTIAL = "ERROR_INVALID_CREDENTIAL";
   private static final String ERROR_RUNTIME_EXCEPTION = "ERROR_RUNTIME_EXCEPTION";
   private static final String PROJECT_ID_MATCH_MESSAGE =
-      " Make sure the ID token comes from the same Firebase project as the service account used to "
+      " Make sure the %s comes from the same Firebase project as the service account used to "
           + "authenticate this SDK.";
-  private static final String VERIFY_ID_TOKEN_DOCS_MESSAGE =
-      " See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to "
-          + "retrieve an ID token.";
+  private static final String VERIFY_TOKEN_DOCS_MESSAGE =
+      " See %s for details on how to retrieve %s.";
   private static final String ALGORITHM = "RS256";
-  private String projectId;
-  private GooglePublicKeysManager publicKeysManager;
 
-  protected FirebaseTokenVerifier(Builder builder) {
-    super(builder);
-    Preconditions.checkArgument(builder.projectId != null, "projectId must be set");
+  private final String projectId;
+  private final GooglePublicKeysManager publicKeysManager;
+  private final String method;
+  private final String shortName;
+  private final String articledShortName;
 
+  private final String projectIdMatchMessage;
+  private final String verifyTokenMessage;
+
+  private FirebaseTokenVerifier(Builder builder) {
+    super(builder);
+    checkArgument(!Strings.isNullOrEmpty(builder.projectId), "projectId must be set");
+    checkArgument(!Strings.isNullOrEmpty(builder.shortName), "shortName must be set");
+    checkArgument(!Strings.isNullOrEmpty(builder.method), "method must be set");
     this.projectId = builder.projectId;
-    this.publicKeysManager = builder.publicKeysManager;
+    this.shortName = builder.shortName;
+    if ("aeiouAEIOU".indexOf(shortName.charAt(0)) < 0) {
+      this.articledShortName = "a " + shortName;
+    } else {
+      this.articledShortName = "an " + shortName;
+    }
```

Comment:
Please make this a helper method. There are also lots of exceptions to these rules... but let's not get into that.

Reply:
Done